api/src/main/java/org/openmrs/obs/handler/AbstractHandler.java

```diff
@@ -119,6 +121,35 @@ public String getExtension(String filename) {
 		return extension;
 	}
 	
+	public String getMimeTypeFromFile(File file) {
+		if (file == null) {
+			return null;
+		}
```

Comment:
`getMimeTypeFromFile` is more complex since it tries to parse MIME type from filename, then if there is no MIME type in file it tries to probe it and returns default MIME type (`application/octet-stream`) if it couldn't guess MIME by any way mentioned before.
`getFileMimeType` is just trying to guess MIME type by using `MimetypesFileTypeMap.getContentType()` which is not enough and it is less reliable than `Files.probeContentType()`